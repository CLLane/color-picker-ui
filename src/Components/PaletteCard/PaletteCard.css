.palette-div {
  display: flex;
  justify-content: center;
}