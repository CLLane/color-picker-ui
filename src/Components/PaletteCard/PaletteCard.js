import React from 'react';
import './PaletteCard.css';


export const PaletteCard = ({ palette, trashPalette }) => {
  const colors = Object.values(palette).slice(2)
  const swatch = colors.map((hex, index) => {
    const divStyle = {
      background: hex,
      height: '5vh',
      width: '5vh',
    }
    return <div style={divStyle} key={index}></div>
  })

  return (
    <div>
      <h4>{ palette.name }</h4>
      <div>
        {swatch}
      </div>
      <button onClick={() => trashPalette(palette.id)}>Trash Can</button>
    </div>
  )
}

export default PaletteCard;