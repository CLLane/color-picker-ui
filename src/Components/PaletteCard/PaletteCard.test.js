import React from 'react';
import { shallow } from 'enzyme';
import PaletteCard from './PaletteCard';
import './PaletteCard.css';

describe('Palette Card', () => {
  let wrapper;
  let mockPalette;
  let mockTrashPalette;
  let mockGrabPalette;
  let mockShowPalette;
  beforeEach(() => {
    mockPalette = {
      id: 1, 
      name: 'Cool Colors',
      project_id: '12345',
      colorOne: '#FFFFFF',
      colorTwo: '#FFFFFF',
      colorThree: '#FFFFFF',
      colorFour: '#FFFFFF',
      colorFive: '#FFFFFF'
    };
    mockTrashPalette = jest.fn();
    mockGrabPalette = jest.fn();
    mockShowPalette = jest.fn();
    wrapper = shallow (<PaletteCard 
      palette={mockPalette}
      trashPalette={mockTrashPalette}
      grabPalette={mockGrabPalette}
      showPalette={mockShowPalette}
    />);
  });

  it('should match the snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('should allow the user to edit the name of the palette', () => {

  });

  it('should allow the use to save the name of palette', () => {

  });

  it('should update the palette name with what a user entered', () => {
    const mockEvent = {
      target: {
        value: 'amazing palette'
      }
    };
    wrapper.instance().handleChange(mockEvent);
    expect(wrapper.state('nameInput')).toEqual('amazing palette');
  });


});