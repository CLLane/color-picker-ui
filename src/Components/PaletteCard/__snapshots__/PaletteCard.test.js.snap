// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Palette Card should match the snapshot 1`] = `
<div
  className="palette__container"
>
  <div
    className="palette-name__container"
  >
    <input
      className="palette-edit__input"
      disabled={true}
      maxlength="13"
      onChange={[Function]}
      type="text"
      value="Cool Colors"
    />
    <img
      alt="edit palette"
      onClick={[Function]}
      src="editIcon.svg"
    />
  </div>
  <div
    className="swatch__container"
  >
    <div
      key="0"
      style={
        Object {
          "background": "#FFFFFF",
          "borderRadius": "8px",
          "height": "5vh",
          "marginLeft": "1.5px",
          "marginRight": "1.5px",
          "marginTop": "2px",
          "width": "5vh",
        }
      }
    />
    <div
      key="1"
      style={
        Object {
          "background": "#FFFFFF",
          "borderRadius": "8px",
          "height": "5vh",
          "marginLeft": "1.5px",
          "marginRight": "1.5px",
          "marginTop": "2px",
          "width": "5vh",
        }
      }
    />
    <div
      key="2"
      style={
        Object {
          "background": "#FFFFFF",
          "borderRadius": "8px",
          "height": "5vh",
          "marginLeft": "1.5px",
          "marginRight": "1.5px",
          "marginTop": "2px",
          "width": "5vh",
        }
      }
    />
    <div
      key="3"
      style={
        Object {
          "background": "#FFFFFF",
          "borderRadius": "8px",
          "height": "5vh",
          "marginLeft": "1.5px",
          "marginRight": "1.5px",
          "marginTop": "2px",
          "width": "5vh",
        }
      }
    />
    <div
      key="4"
      style={
        Object {
          "background": "#FFFFFF",
          "borderRadius": "8px",
          "height": "5vh",
          "marginLeft": "1.5px",
          "marginRight": "1.5px",
          "marginTop": "2px",
          "width": "5vh",
        }
      }
    />
    <Link
      to="/"
    >
      <img
        alt="expand palette"
        onClick={[Function]}
        src="expandIcon.svg"
      />
    </Link>
    <img
      alt="expand palette"
      onClick={[Function]}
      src="expandIcon.svg"
    />
    <img
      alt="delete palette"
      onClick={[Function]}
      src="trashIcon.svg"
    />
  </div>
</div>
`;

exports[`Palette Card should match the snapshot when it is not disabled 1`] = `
<div
  className="palette__container"
>
  <div
    className="palette-name__container"
  >
    <input
      className="palette-edit__input"
      disabled={false}
      maxlength="13"
      onChange={[Function]}
      type="text"
      value="Cool Colors"
    />
    <img
      alt="save icon"
      onClick={[Function]}
      src="saveIcon.svg"
    />
  </div>
  <div
    className="swatch__container"
  >
    <div
      key="0"
      style={
        Object {
          "background": "#FFFFFF",
          "borderRadius": "8px",
          "height": "5vh",
          "marginLeft": "1.5px",
          "marginRight": "1.5px",
          "marginTop": "2px",
          "width": "5vh",
        }
      }
    />
    <div
      key="1"
      style={
        Object {
          "background": "#FFFFFF",
          "borderRadius": "8px",
          "height": "5vh",
          "marginLeft": "1.5px",
          "marginRight": "1.5px",
          "marginTop": "2px",
          "width": "5vh",
        }
      }
    />
    <div
      key="2"
      style={
        Object {
          "background": "#FFFFFF",
          "borderRadius": "8px",
          "height": "5vh",
          "marginLeft": "1.5px",
          "marginRight": "1.5px",
          "marginTop": "2px",
          "width": "5vh",
        }
      }
    />
    <div
      key="3"
      style={
        Object {
          "background": "#FFFFFF",
          "borderRadius": "8px",
          "height": "5vh",
          "marginLeft": "1.5px",
          "marginRight": "1.5px",
          "marginTop": "2px",
          "width": "5vh",
        }
      }
    />
    <div
      key="4"
      style={
        Object {
          "background": "#FFFFFF",
          "borderRadius": "8px",
          "height": "5vh",
          "marginLeft": "1.5px",
          "marginRight": "1.5px",
          "marginTop": "2px",
          "width": "5vh",
        }
      }
    />
    <Link
      to="/"
    >
      <img
        alt="expand palette"
        onClick={[Function]}
        src="expandIcon.svg"
      />
    </Link>
    <img
      alt="expand palette"
      onClick={[Function]}
      src="expandIcon.svg"
    />
    <img
      alt="delete palette"
      onClick={[Function]}
      src="trashIcon.svg"
    />
  </div>
</div>
`;
